### ISSUE TITLE
Sugarcane Leaf Disease Detection

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#456]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Sugarcane Leaf Disease Detection</br>
:red_circle: **Aim** : The aim of this project is to detect the disease from the sugarcane leaves using computer vision and deep learning methods.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/nirmalsankalana/sugarcane-leaf-disease-dataset</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by Aditibusan** [comment]
Hi , I'm excited to contribute to this project through GSSOC'24. Could you please assign me? Looking forward to getting started! @abhisheks008

**Comment by tushtithakur** [comment]
Hi , I'm excited to contribute to this project. Could you please assign me? Looking forward to getting started! @abhisheks008

Full name : Tushti Thakur
GitHub Profile Link : https://github.com/tushtithakur
Email ID : [tushtithakur1234@gmail.com](mailto:tushtithakur1234@gmail.com)
Approach for this Project : Implement different deep learning algorithms using the dataset, evaluate it and compare performance.
What is your participant role? GSSoC 2024

**Comment by Basma2423** [comment]
Wanna give it a try, can you assign it to me? @abhisheks008 

Full name: Basma Mahmoud
GitHub Profile Link: github.com/Basma2423
Email ID: mayarbasma2423@gmail.com
Approach for this Project: Multiple Deep Learning pre-trained models + CNN
What is your participant role? (Mention the Open Source program): GSSoC-2024 participant

Can you add the label for GSSoC, please?
Thanks.

**Comment by atharv1707** [comment]
Hey! I am interested in this project. I have worked on similar project in the past, and I would like to apply some new techniques that I have learned in image classification problems. 

Full name: Atharv Pal 
GitHub Profile Link: https://github.com/atharv1707
Email ID: atharvpal17@gmail.com
Approach for this Project: 

1. Data Preprocessing

- Resize images to a consistent size suitable for deep learning models.
- Normalize pixel values to the range [0, 1].
- Augment the dataset to increase diversity and robustness, because the data set does seem small, and we can increase it.

2. Model Selection

- I am planning to consider architectures like Convolutional Neural Networks (CNNs) due to their effectiveness in image classification.

3. Model Training

- Compile the chosen model with appropriate loss function and optimizer.
- Train the model on the training set and validate its performance on the validation set.
- Fine-tune hyperparameters such as learning rate, batch size, and number of epochs to optimize model performance.
- Monitor training progress using metrics like accuracy, loss, and validation accuracy.

4. Model Evaluation

- We can Evaluate the trained model on the test set to assess its generalization performance.
- Also we can calculate evaluation metrics such as accuracy, precision, recall, and F1-score for each disease category.

5. Model Interpretation and Improvement
Interpret the model's predictions to understand its behavior and identify areas of improvement such as Class Activation Maps (CAM)


What is your participant role?  GSSoC-2024 Contributor

Thanks man!

**Comment by abhisheks008** [comment]
Hi @atharv1707 try to implement different algorithms (atleast 2-3) for this dataset. Compare them based on the accuracy scores and find out the best fitted model for this dataset/project.

Are you up?

**Comment by atharv1707** [comment]
> Hi @atharv1707 try to implement different algorithms (atleast 2-3) for this dataset. Compare them based on the accuracy scores and find out the best fitted model for this dataset/project.
> 
> Are you up?

@abhisheks008  sure mate! I would love to experiment on this!

**Comment by atharv1707** [comment]
Hey @abhisheks008 ! I hope you are doing well. I wanted to update regarding the project i have chosen. I have trained one model using CNN architecture, which gave a accuracy of near 60-65%. I wanted to know what other techniques/algorithms should i implement as you mentioned above.  Personally, i was thinking of experimenting with ensemble techniques to see how much of increament in  accuracy we can achieve.

Peace!

**Comment by abhisheks008** [comment]
> Hey @abhisheks008 ! I hope you are doing well. I wanted to update regarding the project i have chosen. I have trained one model using CNN architecture, which gave a accuracy of near 60-65%. I wanted to know what other techniques/algorithms should i implement as you mentioned above. Personally, i was thinking of experimenting with ensemble techniques to see how much of increament in accuracy we can achieve. If it's alright , I can proceed with the ensemble , or any suggestions are welcome
> 
> Peace!

You can go ahead with your approach. Let's see the accuracy score then I'll suggest something from my end.

**Comment by Pranali3103** [comment]
Hello @abhisheks008 Can you please assign me this issue , I can help with the trying other algorithms.

**Comment by atharv1707** [comment]
Hey @abhisheks008 ! I know it took alot longer that it should have, but I finally did what I aimed to do. I achieved an accuracy of 82.59% using ensemble learning in this project. Kindly let me know what should I do further.

**Comment by abhisheks008** [comment]
> Hey @abhisheks008 ! I know it took alot longer that it should have, but I finally did what I aimed to do. I achieved an accuracy of 82.59% using ensemble learning in this project. Kindly let me know what should I do further.

Need to acheive at least 90%

**Comment by atharv1707** [comment]
> > Hey @abhisheks008 ! I know it took alot longer that it should have, but I finally did what I aimed to do. I achieved an accuracy of 82.59% using ensemble learning in this project. Kindly let me know what should I do further.
> 
> Need to achieve at least 90%

Hey!! @abhisheks008  , I applied ensemble learning and achieved an ensemble accuracy of 90.55%. Let me know what's our next step

**Comment by abhisheks008** [comment]
> > > Hey @abhisheks008 ! I know it took alot longer that it should have, but I finally did what I aimed to do. I achieved an accuracy of 82.59% using ensemble learning in this project. Kindly let me know what should I do further.
> > 
> > 
> > Need to achieve at least 90%
> 
> Hey!! @abhisheks008 , I applied ensemble learning and achieved an ensemble accuracy of 90.55%. Let me know what's our next step

Push that code for review.
